### PR TITLE
Clean up TextTokenizer

### PR DESCRIPTION
**Pull request by zechengz**
Fix https://github.com/pyg-team/pytorch-frame/issues/216

### CODE REVIEW

File: test/datasets/test_fake_dataset.py [hunk 1]
```diff
@@ -11,7 +11,8 @@
 
 
 @pytest.mark.parametrize('with_nan', [True, False])
-def test_fake_dataset(with_nan):
+@pytest.mark.parametrize('tokenize_with_batched', [True, False])
```
Comment (weihua916) [inline test/datasets/test_fake_dataset.py:14]: tokenize_with_batch

File: torch_frame/typing.py [hunk 2]
```diff
@@ -63,5 +63,6 @@ def is_numerical_strategy(self):
 
 IndexSelectType = Union[int, List[int], range, slice, Tensor]
 ColumnSelectType = Union[str, List[str]]
-TextTokenizationOutputs = List[Dict[str, Tensor]]
+TextTokenizationOutputs = Union[List[Mapping[str, Tensor]], Mapping[str,
+                                                                    Tensor]]
```
Comment (weihua916) [inline torch_frame/typing.py:67]: super nit: awkward line breaking. Maybe add parenthesis.

File: torch_frame/data/mapper.py [hunk 3]
```diff
@@ -341,15 +343,19 @@ def forward(
...
-        for key in tokenized_list[0]:
+            if isinstance(tokenized_batch, Mapping):
+                tokenized_outputs.append(tokenized_batch)
+            else:
+                tokenized_outputs.extend(tokenized_batch)
+
+        for key in tokenized_outputs[0]:
             xs = []
-            for item in tokenized_list:
+            for item in tokenized_outputs:
```
Comment (weihua916) [inline torch_frame/data/mapper.py:358]: Can we branch 
```python
                if item[key].ndim == 1:
                    xs.append([item[key]])
                elif item[key].ndim == 2:
                    for row in item[key]:
                        xs.append([row])
```
based on 
```python
if isinstance(tokenized_batch, Mapping):
```
and then assert on ndim?

Comment (weihua916) [inline torch_frame/data/mapper.py:358]: `for tokenized_batch in tokenized_outputs`
to align with your previous for-loop

Comment (zechengz) [reply torch_frame/data/mapper.py:358]: Do you mean if the `tokenized_batch` is `Mapping`, tensors should be dim=2?

File: torch_frame/data/mapper.py [hunk 4]
```diff
@@ -328,43 +328,38 @@ def forward(
...
+                for tokenized_batch in tokenized_outputs:
+                    xs.extend([row] for row in tokenized_batch[key])
+                feat_dict[key] = MultiNestedTensor.from_tensor_mat(xs)
+        else:
+            keys = tokenized_outputs[0][0].keys()
+            for key in keys:
+                xs = []
+                for tokenized_batch in tokenized_outputs:
+                    xs.extend([item[key]] for item in tokenized_batch)
+                feat_dict[key] = MultiNestedTensor.from_tensor_mat(xs)
```
Comment (weihua916) [inline torch_frame/data/mapper.py:362]: nit: the following can be put outside of if-else statement.
```python
feat_dict[key] = MultiNestedTensor.from_tensor_mat(xs)
```

Comment (weihua916) [inline torch_frame/data/mapper.py:362]: Can we raise error on `ndim` of `tokenized_batch[key]`?

Comment (zechengz) [reply torch_frame/data/mapper.py:362]: I think for each key we need to call `from_tensor_mat` and each type of `tokenized_outputs` (mapping or list) need to get `keys` in a different way. So I think it still need to be put inside the if else.

Comment (weihua916) [reply torch_frame/data/mapper.py:373]: Oh I see. Thanks for clarifying.

**Review by weihua916 — APPROVED** [review]
Very clean, thanks!

**Pull request merged**